airflow/models/taskinstance.py

```diff
@@ -2709,43 +3004,12 @@ def _execute_task(self, context: Context, task_orig: Operator):
         return _execute_task(self, context, task_orig)
 
     @provide_session
-    def defer_task(self, session: Session, defer: TaskDeferred) -> None:
+    def defer_task(self, exception: TaskDeferred, session: Session) -> None:
```

Comment:
Out of curiosity why renaming `defer` to `exception`?

Reply:
because defer doesn't really have an obvious meaning.... i could do this in a separate PR. i figured it's ok cus i think it's marked :meta private: